roxctl/declarativeconfig/create/notifier.go

```diff
@@ -0,0 +1,240 @@
+	notifierCommand := &notifierCmd{notifier: &declarativeconfig.Notifier{}, env: cliEnvironment}
+
+	cmd := &cobra.Command{
+		Use:   notifierCommand.notifier.Type(),
+		Args:  cobra.NoArgs,
+		Short: "Commands to create a declarative configuration for a notifier",
+	}
+
+	cmd.PersistentFlags().StringVar(&notifierCommand.notifier.Name, "name", "", "Name of the notifier")
+	cmd.AddCommand(
+		notifierCommand.genericCommand(),
+		notifierCommand.splunkCommand())
+
+	utils.Must(cmd.MarkPersistentFlagRequired("name"))
+	return cmd
+}
+
+func (n *notifierCmd) genericCommand() *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:   "generic",
+		Args:  cobra.NoArgs,
+		RunE:  n.runE(),
+		Short: "Create a declarative configuration for a generic notifier",
+	}
+	genericFlags := cmd.Flags()
+	n.gc = &declarativeconfig.GenericConfig{}
+	genericFlags.BoolVar(&n.gc.AuditLoggingEnabled, "audit-logging", false,
+		"Audit logging enabled")
+	genericFlags.StringVar(&n.gc.Endpoint, "webhook-endpoint", "",
+		"Webhook endpoint URL")
+	genericFlags.StringVar(&n.gc.Username, "webhook-username", "",
+		"Username for the webhook endpoint basic authentication. "+
+			"No authentication if not provided. Requires --webhook-password")
+	genericFlags.StringVar(&n.gc.Password, "webhook-password", "",
+		"Password for the webhook endpoint basic authentication. "+
+			"No authentication if not provided. Requires --webhook-username")
+	genericFlags.StringVar(&n.gc.CACertPEM, "webhook-cacert-file", "",
+		"Endpoint CA certificate file name (PEM format)")
+	genericFlags.StringToStringVar(&n.gcHeaders, "headers", nil,
+		"Headers (comma separated key=value pairs)")
+	genericFlags.StringToStringVar(&n.gcExtraFields, "extra-fields", nil,
+		"Extra fields (comma separated key=value pairs)")
+	genericFlags.BoolVar(&n.gc.SkipTLSVerify, "webhook-insecure-skip-tls-verify", false,
+		"Skip webhook TLS verification")
+	n.genericFlagSet = genericFlags
+
+	cmd.Flags().AddFlagSet(genericFlags)
+
+	cmd.MarkFlagsRequiredTogether("webhook-username", "webhook-password")
+	utils.Must(cmd.MarkFlagFilename("webhook-cacert-file"))
+	utils.Must(cmd.MarkFlagRequired("webhook-endpoint"))
+	return cmd
+}
+
+func (n *notifierCmd) splunkCommand() *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:   "splunk",
+		Args:  cobra.NoArgs,
+		RunE:  n.runE(),
+		Short: "Create a declarative configuration for a Splunk notifier",
+	}
+
+	splunkFlags := cmd.Flags()
+	n.sc = &declarativeconfig.SplunkConfig{}
+	splunkFlags.BoolVar(&n.sc.AuditLoggingEnabled, "audit-logging", false,
+		"Audit logging enabled")
+	splunkFlags.StringVar(&n.sc.HTTPToken, "splunk-token", "",
+		"Splunk HTTP token (required)")
+	splunkFlags.StringVar(&n.sc.HTTPEndpoint, "splunk-endpoint", "",
+		"Splunk HTTP endpoint (required)")
+	splunkFlags.BoolVar(&n.sc.Insecure, "splunk-insecure-skip-tls-verify", false,
+		"Insecure connection to Splunk")
+	splunkFlags.Int64Var(&n.sc.Truncate, "truncate", 0,
```

Comment:
Maybe let's use `10000` as it is default limit in the UI?

Reply:
I'll add this to the description, but keep 0 as the default value so that the key doesn't appear in the output (`json: omitempty`) when not set.